builtin/common/vector.lua

```diff
@@ -21,12 +21,22 @@ function metatable.__newindex(v, key, value)
 	rawset(v, xyz[key] or key, value)
 end
 
+local namespace_metatable = {}
+setmetatable(vector, namespace_metatable)
+
 -- constructors
 
 local function fast_new(x, y, z)
 	return setmetatable({x = x, y = y, z = z}, metatable)
 end
 
+function namespace_metatable.__call(_, x, y, z)
+	if x and y and z then
+		return fast_new(x, y, z)
+	end
+	error("Invalid arguments passed to vector()")
+end
```

Comment:
`error(…, 2)` should be used here, so the trace-back points to where `vector()` was called, not where `error()` was called.
Whilst you're at it, change the `assert` below to an error doing the same.
`vector.copy` could do with the same "optimization" of `x and y and z`.

Unfortunately GitHub won't allow me to make a suggestion including lines not within your commit :facepalm:, so:
```diff
diff --git a/builtin/common/vector.lua b/builtin/common/vector.lua
index 4bbfd37..a5f7cd5 100644
--- a/builtin/common/vector.lua
+++ b/builtin/common/vector.lua
@@ -36,3 +36,3 @@ function namespace_metatable.__call(_, x, y, z)
 	end
-	error("Invalid arguments passed to vector()")
+	error("Invalid arguments passed to vector()", 2)
 end
@@ -47,4 +47,5 @@ function vector.new(a, b, c)
 		return vector.copy(a)
+	elseif a or b or c then
+		error("Invalid arguments passed to vector.new()", 2)
 	else
-		assert(not a, "Invalid arguments for vector.new()")
 		return vector.zero()
@@ -58,4 +59,7 @@ end
 function vector.copy(v)
-	assert(v.x and v.y and v.z, "Invalid vector passed to vector.copy()")
-	return fast_new(v.x, v.y, v.z)
+	if v.x and v.y and v.z then
+		return fast_new(v.x, v.y, v.z)
+	else
+		error("Invalid vector passed to vector.copy()", 2)
+	end
 end
```